sql/tests: stress: TestMonotonicInserts failed due to contention

sql/tests.TestMonotonicInserts [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7717680?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7717680?buildTab=artifacts#/) on release-22.2 @ [fd133ba3af3954c93584a501cf68dfa3131e586c](https://github.com/cockroachdb/cockroach/commits/fd133ba3af3954c93584a501cf68dfa3131e586c):


```
=== RUN   TestMonotonicInserts
--- FAIL: TestMonotonicInserts (18.79s)
=== RUN   TestMonotonicInserts/distsql=off
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestMonotonicInserts_distsql=off2833529335
    test_log_scope.go:79: use -show-logs to present logs inline
    monotonic_insert_test.go:186: *crdb.MaxRetriesExceededError: retrying txn failed after 50 attempts. original error: pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE - failed preemptive refresh): "sql txn" meta={id=baa48bd2 key=/Table/106/1/817509944182505475/0 pri=0.04263162 epo=0 ts=1669554635.892981882,1 min=1669554635.857152263,0 seq=1} lock=true stat=PENDING rts=1669554635.857152263,0 wto=false gul=1669554636.357152263,0.
    monotonic_insert_test.go:239: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestMonotonicInserts_distsql=off2833529335
    --- FAIL: TestMonotonicInserts/distsql=off (9.35s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #92461 sql/tests: TestMonotonicInserts failed [C-test-failure O-robot T-sql-queries branch-master]
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMonotonicInserts.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-21847

sql/tests.TestMonotonicInserts [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7771982?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7771982?buildTab=artifacts#/) on release-22.2 @ [d978988afcfc78c02c23a3836ba90e4a4faeeb25](https://github.com/cockroachdb/cockroach/commits/d978988afcfc78c02c23a3836ba90e4a4faeeb25):


```
=== RUN   TestMonotonicInserts
--- FAIL: TestMonotonicInserts (18.90s)
=== RUN   TestMonotonicInserts/distsql=off
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestMonotonicInserts_distsql=off2124575532
    test_log_scope.go:79: use -show-logs to present logs inline
    monotonic_insert_test.go:186: *crdb.MaxRetriesExceededError: retrying txn failed after 50 attempts. original error: pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE - failed preemptive refresh): "sql txn" meta={id=3cc489b4 key=/Table/106/1/818335221151072259/0 pri=0.07768096 epo=1 ts=1669806490.425892556,1 min=1669806490.337899982,0 seq=1} lock=true stat=PENDING rts=1669806490.340258050,0 wto=false gul=1669806490.837899982,0.
    monotonic_insert_test.go:239: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestMonotonicInserts_distsql=off2124575532
    --- FAIL: TestMonotonicInserts/distsql=off (9.32s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMonotonicInserts.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/tests.TestMonotonicInserts [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7789086?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7789086?buildTab=artifacts#/) on release-22.2 @ [ca673993473629a46370100108dd8c185a592e91](https://github.com/cockroachdb/cockroach/commits/ca673993473629a46370100108dd8c185a592e91):


```
=== RUN   TestMonotonicInserts
--- FAIL: TestMonotonicInserts (16.88s)
=== RUN   TestMonotonicInserts/distsql=off
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestMonotonicInserts_distsql=off3541439087
    test_log_scope.go:79: use -show-logs to present logs inline
    monotonic_insert_test.go:186: *crdb.MaxRetriesExceededError: retrying txn failed after 50 attempts. original error: pq: restart transaction: TransactionRetryWithProtoRefreshError: TransactionRetryError: retry txn (RETRY_SERIALIZABLE - failed preemptive refresh): "sql txn" meta={id=eaaae1d3 key=/Table/106/1/818633939790364675/0 pri=0.04125210 epo=0 ts=1669897652.128290746,1 min=1669897652.100184700,0 seq=1} lock=true stat=PENDING rts=1669897652.100184700,0 wto=false gul=1669897652.600184700,0.
    monotonic_insert_test.go:239: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestMonotonicInserts_distsql=off3541439087
    --- FAIL: TestMonotonicInserts/distsql=off (8.37s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMonotonicInserts.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/tests/monotonic_insert_test.go
+++ b/pkg/sql/tests/monotonic_insert_test.go
@@ -94,6 +94,7 @@ func testMonotonicInserts(t *testing.T, distSQLMode sessiondatapb.DistSQLExecMod
 
 	skip.UnderShort(t)
 	skip.UnderRace(t) // Too slow under race.
+	skip.UnderStressWithIssue(t, 92540, "too much contention under stress")
 
 	ctx := context.Background()
 	tc := testcluster.StartTestCluster(

--- a/pkg/testutils/skip/skip.go
+++ b/pkg/testutils/skip/skip.go
@@ -119,6 +119,17 @@ func UnderStress(t SkippableTest, args ...interface{}) {
 	}
 }
 
+// UnderStressWithIssue skips this test when running under stress, logging the
+// given issue ID as the reason.
+func UnderStressWithIssue(t SkippableTest, githubIssueID int, args ...interface{}) {
+	t.Helper()
+	if Stress() {
+		t.Skip(append([]interface{}{fmt.Sprintf(
+			"disabled under stress. issue: https://github.com/cockroachdb/cockroach/issues/%d", githubIssueID,
+		)}, args...))
+	}
+}
+
 // UnderStressRace skips this test during stressrace runs, which are tests
 // run under stress with the -race flag.
 func UnderStressRace(t SkippableTest, args ...interface{}) {

